---
title: "compare aligning reads to whole genome"
output: html_notebook
---

load packages

```{r}
library(ggplot2)
library(data.table)
library(stringr)
library(dplyr)
library(tidyr)
library(rprojroot)
library(vegan)
library(ggpubr)
library(rstatix)
library(cowplot)
library(forcats)

```

set paths and filenames

```{r}
### files
long_table=sprintf("%s/data/mp142_TGVG1.1_MPA4_combined_abundance_table_longform1.tsv",
                   find_rstudio_root_file())
metadata_table=sprintf("%s/data/some_teddy_MP142_metadata2.all_samples1.delivery.csv",
                       find_rstudio_root_file())

coverm_table=sprintf("%s/data/all_shuf_samples_read_align.tsv", 
                        find_rstudio_root_file())

marker_magu_features=sprintf("%s/data/Marker-MAGu_virus_DB_v1.1_metadata.tsv",
                             find_rstudio_root_file())

```

load tables
```{r}
cover_dt <- fread(coverm_table, header = F, 
                  col.names = c("sampleID", "contig_id", "contig_length",
                                "covered_bases", "reads_aligned")) %>%
  mutate(aligned_proportion = covered_bases / contig_length) %>%
  filter(aligned_proportion >= 0.1)
```

```{r}
long_dt <- fread(long_table, sep = "\t", header = T) %>%
  select(sampleID, rel_abundance, lineage) %>%
  mutate(sampleID = as.character(sampleID),
         kingdom = case_when(grepl("k__Bac", lineage) ~ "Bacteria", 
                             grepl("k__Vir", lineage) ~ "Virus",
                             grepl("k__Ar", lineage) ~ "Archea",
                             grepl("k__Euk", lineage) ~ "Eukaryota",
                             TRUE ~ "other"),
         species = gsub(".*s__","s__", lineage))

mm_features_dt <- fread(marker_magu_features, header = T, sep = "\t")
```

```{r}
subsample_dt <- long_dt %>%
  filter(sampleID %in% unique(cover_dt$sampleID)) 
```

```{r}
cover_meta_dt <- merge(cover_dt, mm_features_dt %>% select(-contig_length),
                      by = "contig_id") %>%
  mutate(sampleID = as.character(sampleID),
         lineage = gsub("s__vSGB", "s__vOTU", lineage),
         species = gsub(".*s__","s__", lineage))
```

```{r}
match_dt <- merge(subsample_dt%>% select(-c(lineage, kingdom)), 
                  cover_meta_dt,
      by = c("sampleID", "species"),
      all.y = T) %>%
  filter(`marker-magu_genes` >= 4) %>%
  mutate(detected = case_when(is.na(rel_abundance) ~ "Not-detected",
                              TRUE ~ "Detected")) %>%
  group_by(contig_id, sampleID) %>%
  mutate(any_detect = case_when(any(detected %in% "Detected") ~ "Detected",
                              TRUE ~ "not-detected")) %>%
  mutate(rel_abundance = case_when(is.na(rel_abundance) ~ 0,
                                   TRUE ~ rel_abundance)) %>%
  ungroup() %>%
  mutate(cov_bins = cut(aligned_proportion*100, 
                    breaks = c(10, 20, 30, 40, 50, 60, 70, 80, 90, 100)))
```

```{r}
match_dt %>%
  filter(!is.na(cov_bins)) %>%
  group_by(cov_bins, detected) %>%
  summarize(n = n()) %>%
  mutate(prop = prop.table(n)) %>%
  ggplot(aes(x = cov_bins, y = prop, fill = fct_reorder(detected, n))) +
  geom_col() +
  scale_fill_brewer(palette = 7,
                    name = "Status") +
  scale_y_continuous(labels = scales::percent) +
  labs(x = "Bins: Percent genomes covered by reads",
       y = "Detection") +
  theme_minimal()
```

```{r}
cover_all_dt <- fread(coverm_table, header = F, 
                  col.names = c("sampleID", "contig_id", "contig_length",
                                "covered_bases", "reads_aligned")) %>%
  mutate(aligned_proportion = covered_bases / contig_length)

cover_all_meta_dt <- merge(cover_all_dt, mm_features_dt %>% select(-contig_length),
                      by = "contig_id") %>%
  mutate(sampleID = as.character(sampleID),
         lineage = gsub("s__vSGB", "s__vOTU", lineage),
         species = gsub(".*s__","s__", lineage))
```

```{r}
detect_match <- merge(subsample_dt, 
                  cover_all_meta_dt %>% select(-c(lineage, iphop_host_genus,
                                                  iphop_confidence, Virulence_score,
                                                  checkv_completeness)),
      by = c("sampleID", "species"),
      all.x = T) %>%
  filter(kingdom == "Virus") %>%
  mutate(#aligned_status = case_when(is.na(reads_aligned) ~ "Not-aligned",
        #                      TRUE ~ "Aligned"),
         aligned_proportion = case_when(is.na(reads_aligned) ~ 0,
                                   TRUE ~ aligned_proportion))
```

```{r}
detect_match %>%
  ggplot(aes(x = rel_abundance,
             y = aligned_proportion)) +
  geom_point()
```

```{r}
detect_match %>%
  ggplot(aes(x = aligned_proportion)) +
  geom_histogram(bins = 50, fill = "grey15") +
  theme_bw() +
  geom_vline(xintercept = median(detect_match$aligned_proportion)) +
  #geom_vline(xintercept = mode(detect_match$aligned_proportion)[[1]]) +
  geom_vline(xintercept = mean(detect_match$aligned_proportion),
             color = "blue")
  

```




